include/pybind11/numpy.h

```diff
@@ -84,37 +120,33 @@ class array : public buffer {
 
     template <typename Type> array(size_t size, const Type *ptr) {
         API& api = lookup_api();
-        PyObject *descr = api.PyArray_DescrFromType_(detail::npy_format_descriptor<Type>::value);
-        if (descr == nullptr)
-            pybind11_fail("NumPy: unsupported buffer format!");
+        PyObject *descr = detail::npy_format_descriptor<Type>::dtype().release().ptr();
         Py_intptr_t shape = (Py_intptr_t) size;
         object tmp = object(api.PyArray_NewFromDescr_(
             api.PyArray_Type_, descr, 1, &shape, nullptr, (void *) ptr, 0, nullptr), false);
-        if (ptr && tmp)
-            tmp = object(api.PyArray_NewCopy_(tmp.ptr(), -1 /* any order */), false);
         if (!tmp)
             pybind11_fail("NumPy: unable to create array!");
+        if (ptr)
+            tmp = object(api.PyArray_NewCopy_(tmp.ptr(), -1 /* any order */), false);
         m_ptr = tmp.release().ptr();
     }
 
     array(const buffer_info &info) {
-        API& api = lookup_api();
-        if ((info.format.size() < 1) || (info.format.size() > 2))
-            pybind11_fail("Unsupported buffer format!");
-        int fmt = (int) info.format[0];
-        if (info.format == "Zd")      fmt = API::NPY_CDOUBLE_;
-        else if (info.format == "Zf") fmt = API::NPY_CFLOAT_;
-
-        PyObject *descr = api.PyArray_DescrFromType_(fmt);
-        if (descr == nullptr)
-            pybind11_fail("NumPy: unsupported buffer format '" + info.format + "'!");
+        auto& api = lookup_api();
+
+        // _dtype_from_pep3118 returns dtypes with padding fields in, however the array
+        // constructor seems to then consume them, so we don't need to strip them ourselves
```

Comment:
I vaguely remember commenting about this but can't find it anymore. Basically the comment above makes me scratch my head: it says "we *don't* need to strip [.. padding fields ..]" but then the code below calls a strip_padding_fields function. Maybe just a wording issue?